pkg/controller/statusmanager/pod_status.go

```diff
@@ -91,6 +92,21 @@ func (status *StatusManager) SetFromPods() {
 		} else if ds.Status.NumberUnavailable > 0 {
 			progressing = append(progressing, fmt.Sprintf("DaemonSet %q is not available (awaiting %d nodes)", dsName.String(), ds.Status.NumberUnavailable))
 			dsProgressing = true
+			// Check for any pods in CrashLoopBackOff state and mark the operator as degrade if so.
+			pods := &v1.PodList{}
+			err := status.client.List(context.TODO(), pods, client.InNamespace(dsName.Namespace), client.MatchingLabels(ds.Spec.Selector.MatchLabels))
```

Comment:
suggestion: refactor this in to a function that takes namespace and label selector.

Reply:
yea, good point, was thinking the same to avoid code duplication, but just wanted a initial code screen first to see how the CI behaves.

Reply:
I also have to add tests